test/e2e/performanceprofile/functests/utils/cgroup/v2/v2.go

```diff
@@ -60,6 +60,32 @@ func (cm *ControllersManager) Cpu(ctx context.Context, pod *corev1.Pod, containe
 	return cfg, nil
 }
 
+func (cm *ControllersManager) CpuStat(ctx context.Context, pod *corev1.Pod, containerName, childName, runtimeType string) (*controller.CpuStat, error) {
+	cfg := &controller.CpuStat{}
+	cfg.Stat = make(map[string]string)
+	dirPath := path.Join(controller.CgroupMountPoint, childName)
+	cmd := []string{
+		"/bin/cat",
+		dirPath + "/cpu.stat",
+	}
+	b, err := pods.ExecCommandOnPod(cm.k8sClient, pod, containerName, cmd)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve cgroup config for pod. pod=%q, container=%q; %w", client.ObjectKeyFromObject(pod).String(), containerName, err)
+	}
+	output := strings.TrimSpace(string(b))
+	interfacevalues := strings.Split(output, "\r\n")
+	// cpu.stat always contains 3 stats usage_usec, user_usec, system_usec
+	// only when cpu controller is enabled other stats like nr_periods etc are enabled
+	if len(interfacevalues) < 4 {
```

Comment:
A more robust way to check whether cpu controller enabled on cgroup v2 is by checking the `cgroup.controllers` and `cgroup.subtree_control` files.
From the documentation https://man7.org/linux/man-pages/man7/cgroups.7.html:
_Active cgroups must be specified via the files
cgroup.controllers and cgroup.subtree_control._

Reply:
It's only a suggestion, if we want to keep the function identical this is also an option but I think it would be better to move the common (identical for v1 and v2) functions into a separate package.
This can be handled in a separate PR as well

Reply:
Yeah i will address this in separate PR , for now i will modify the message.